src/lib/components/SettingsModal.svelte

```diff
@@ -10,6 +10,8 @@
 
 	export let settings: Pick<Settings, "shareConversationsWithModelAuthors">;
 
+	let shareConversationsWithModelAuthors = settings.shareConversationsWithModelAuthors;
```

Comment:
```suggestion
	$: shareConversationsWithModelAuthors = settings.shareConversationsWithModelAuthors;
```

Maybe? If the data is reloaded for X or Y

Reply:
Ahh actually! this variable represents the local state of the form, it shouldn't get updated by anything external since it represents user's choice while it is configuring its settings. Once user "applied" (ie when modal is closed), then at this point it can be updated.